lib/Form/DisplayFormInWPContent.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace MailPoet\Form;
+
+use MailPoet\Entities\FormEntity;
+use MailPoet\WP\Functions as WPFunctions;
+
+class DisplayFormInWPContent {
+
+  /** @var WPFunctions */
+  private $wp;
+
+  /** @var FormsRepository */
+  private $formsRepository;
+
+  public function __construct(WPFunctions $wp, FormsRepository $formsRepository) {
+    $this->wp = $wp;
+    $this->formsRepository = $formsRepository;
+  }
+
+  // TODO remove transient in the api on form save
+
+  public function display(string $content): string {
+    $result = $content;
+    if (!$this->wp->isSingle()) return $result;
+    $forms = $this->formsRepository->findAll();
+    foreach ($forms as $form) {
+      $result .= $this->getContentBellow($form);
+    }
+    return $result;
+  }
+
+  private function getContentBellow(FormEntity $form): string {
+    if (!$this->shouldDisplayFormBellowContent($form)) return '';
+    return Renderer::render([
+      'body' => $form->getBody(),
+      'styles' => $form->getStyles(),
+    ]);
+  }
+
+  private function shouldDisplayFormBellowContent(FormEntity $form): bool {
+    $settings = $form->getSettings();
+    if (!is_array($settings)) return false;
+    if (!isset($settings['placeFormBellowAllPosts'])) return false;
+    if (
+      ($settings['placeFormBellowAllPosts'] === '1')
+      && !$this->wp->isPage()
+    ) return true;
+    if (
+      ($settings['placeFormBellowAllPages'] === '1')
+      && $this->wp->isPage()
+    ) return true;
```

Comment:
I think this check is not correct. We have two switches: below post and below pages. With this implementation both do the same they will display if is_page returns true. We need somehow distinguish if the page is related to post or if it is a page. I can see that in wp_post table there is a post_type column so maybe we can use that information somehow. I'm not really sure what is the best way of doing it. Maybe we should also consider what to do with custom post types e.g. WooCommerce product. 
If we distinguish types it would probably require to have more caching keys.

Reply:
Hey my bad. I missed the negation in condition `!$this->wp->isPage()` for posts 🤦‍♂